### ISSUE TITLE
NaN loss

### ISSUE THREAD

**Issue by vaishnavm217** [HiKapok/SSD.TensorFlow#22]
Hi,
I have changed the model definition for face detection task. As mentioned in readme, I have changed the required places for adapting it in 1.4tf.
But NaN loss issue is not getting solved. Even in reduced learning rate of 1e-4 its giving Nan.
have you found any solution to this?

**Comment by HiKapok** [comment]
@vaishnavm217 please set learning rate to zero and check again, if the problem still exists, try to futher check the annotions of your face datasets to ensure  all images have at least one face bbox.

**Comment by vaishnavm217** [comment]
> to ensure all images have at least one face bbox.

I have filtered out the wrong annotations and removed the images with no faces.
Although unsure if the preprocessing script would take regions with no faces.
> But NaN loss issue is not getting solved

Forgot to mention, this happens in between training, the model trains and suddenly either loc loss or  ce loss becomes NaN.
I have reduced the checkpoint interval for the same.
The losses are high too (loc_loss: 15 to 7, l2loss=~11), doesn't seem converging much.
> set learning rate to zero and check again

Will try this once the NaN occurs

**Comment by HiKapok** [comment]
@vaishnavm217  your batch size is too small, please drop the learning rate lower.

**Comment by franklu323** [comment]
Sorry, I just reopen this issue again.
Now I can train my own model, but it got nan. I did change the learning rate (0, or lower LR), and also follow the Readme. How can I solve this problem? My batch size is set to 16. @HiKapok

**Comment by HiKapok** [comment]
@franklu323 I think the problem is still caused by your dataset, can you provide more details?

**Comment by vaishnavm217** [comment]
So won't the training take **ETERNITY** time if the learning rate so low?

**Comment by HiKapok** [comment]
@vaishnavm217 you should keep lr/batch_size almost the same while you change batch size

**Comment by HiKapok** [comment]
@franklu323 there are quite a lot ways to figure out the loss NAN [roblem, you can use tf's debug tool which will terminate when NAN loss was encountered. you can also try to set features to all ones(tf.ones_like) to further check if there is problem with datas. But I will recommend you to check the anchor regress targets and classification targets, to ensure both postitive anchors and negative anchors is not zero

**Comment by 365gg** [comment]
I fix this problem by setting multi_gpu=False
